src/lambdaconnect_sync/db_drivers/memory.cljc

```diff
@@ -0,0 +1,729 @@
+                              #(conj (or % #{}) (id->rel entity-id)))))
+
+          ;; one to one
+
+          (and (not (:to-many relationship))
+               (not (:to-many inverse-relationship)))         
+          
+          (let [old-value (get-in snapshot 
+                                  [:collections
+                                   entity-name
+                                   :collection-content
+                                   entity-id
+                                   attribute-key])
+                old-target-value (-> snapshot 
+                                     (get-in [:collections
+                                              (:destination-entity relationship)
+                                              :collection-content
+                                              new-value
+                                              (t/datomic-inverse-name inverse-relationship)])
+                                     (first)
+                                     :db/id)]
+            
+            (cond-> snapshot
+            
+              old-value
+              (update-in [:collections
+                          (:destination-entity relationship)
+                          :collection-content
+                          old-value
+                          (t/datomic-inverse-name inverse-relationship)]
+                         #(disj (or % #{}) (id->rel entity-id)))
+  
+              old-target-value 
+              (update-in [:collections
+                          entity-name
+                          :collection-content
+                          old-target-value]
+                         #(do
+                            (assert (= entity-id (attribute-key %)))
+                            (dissoc % attribute-key)))
+
+              true (assoc-in [:collections
+                              entity-name
+                              :collection-content
+                              entity-id
+                              attribute-key]
+                             (id->rel new-value))
+              
+              true (assoc-in [:collections
+                              (:destination-entity relationship)
+                              :collection-content
+                              new-value
+                              (t/datomic-inverse-name inverse-relationship)]
+                             #{(id->rel entity-id)}))))))))
+
+(defmethod process-transaction-entry :db/retract
+  [[_ entity-id attribute-key retracted-value] 
+   snapshot 
+   datomic-relationships 
+   entities-by-id
+   allocated-ids]
+
+  (let [entity-name (entities-by-id entity-id)]
+    (assert entity-name)
+    (if-not (datomic-relationships attribute-key)
+      ;; Attribute
+      (update-in snapshot [:collections 
+                          entity-name 
+                          :collection-content 
+                          entity-id]
+                 #(do (assert (= retracted-value (attribute-key %)))
```

Comment:
Można napisać też `[:db/retract id attr]` (bez wartości), sprawdź tego asserta tylko jeśli `retracted-value` nie jest nilem/

Reply:
Mój driver tego nie dozwala i widać to też w specu. Wyrażenia które konstruuje synchronizacja zawsze mają wartość, uważam że to jest bezpieczniejsze i nigdy nie powinno się stosować bez wartości.